src/python/WMCore/ReqMgr/Service/Auxiliary.py

```diff
@@ -186,10 +195,41 @@ def post(self):
             raise MissingPostData()
         else:
             doc = json.loads(data)
-        doc = Document(self.name, doc)
+        if subName:
+            docName = "%s_%s" % (self.name, subName)
+        else:
+            docName = self.name
+        doc = Document(docName, doc)
         result = self.reqmgr_aux_db.commitOne(doc)
         return result
 
+    @restcall(formats=[('text/plain', PrettyJSONFormat()), ('application/json', JSONFormat())])
+    def put(self, subName=None):
+        """
+        update document for the given  self.name and subName
+        """
+        data = cherrypy.request.body.read()
+        if not data:
+            raise MissingPostData()
+        else:
+            propertyDict = json.loads(data)
+
+        try:
+            result = None
+            if subName:
+                docName = "%s_%s" % (self.name, subName)
+            else:
+                docName = self.name
+
+            if propertyDict:
```

Comment:
Every property values is required explicitly. So if agent is out of drain, it should have explicitly say {"AgentDrainMode": False} and current code (with this patch) will do that.